presto-main/src/main/java/com/facebook/presto/spiller/BinaryFileSingleStreamSpillerFactory.java

```diff
@@ -80,6 +90,29 @@ private static ListeningExecutorService createExecutorServiceOfSize(int nThreads
         return MoreExecutors.listeningDecorator(executorService);
     }
 
+    @PostConstruct
+    public void cleanupOldSpillFiles()
+    {
+        spillPaths.forEach(this::cleanupOldSpillFiles);
+    }
+
+    private void cleanupOldSpillFiles(Path path)
+    {
+        Arrays.stream(path.toFile().listFiles(new PatternFilenameFilter(SPILL_FILE_PATTERN))).forEach(
```

Comment:
If you really want to stick with `PatternFilenameFilter` you should precompile the pattern in the constant.